### ISSUE TITLE
Scripting request: SAMs and IADS in range 23 (Scenario)

### ISSUE THREAD

**Issue by 132ndNeck** [132nd-vWing/TRMA#38]
@132nd-Entropy:
Would it be possible to script the following for range 23:

**IADS**
**Step 1: The following units need to be activated via F10 menu: (ability to turn on/off. Off= deactivates units)**
R23_EWR_1
R23_EWR_2
R23_EWR_3
R23_IADS_SA2-1
R23_IADS_SA2-2
R23_IADS_SA15_pointdefence_1
R23_IADS_SA15_pointdefence_2

**Step 2: Activate the following mobile SAMs:**
Activate a total of 3 SAM groups from the following templates:  (Need to have R23_IADS to start the name to have the IADS script work)
R23_IADS_SA6_1    
R23_IADS_SA11_1

Activate the 3 SAMs above randomly based on the following locations:
R23_SAM_SPAWN_1
R23_SAM_SPAWN_2
R23_SAM_SPAWN_3
R23_SAM_SPAWN_4
R23_SAM_SPAWN_5
R23_SAM_SPAWN_6
R23_SAM_SPAWN_7
R23_SAM_SPAWN_8
R23_SAM_SPAWN_9

**Step 3: Activate IADS** 
Flag 22 need to be set to true when the units above are activated, as it will activate the IADS for the units listed above

**Comment by 132ndNeck** [comment]
Same as with the other ranges, I have them visible and not late activated until the script is ready, then I will fix it in the mission editor

**Comment by 132ndNeck** [comment]
First post updated with mobile SAMs

**Comment by 132nd-Entropy** [comment]
@132ndNeck 

SAMS in step 1 should all be spawned in with the script via F10 menu? all wit AI on ? and then have a toggle for each of them to turn on/off the AI ? 

Step 2 you write 3 sams but there only 2 templates?

**Comment by 132ndNeck** [comment]
> SAMS in step 1 should all be spawned in with the script via F10 menu? all wit AI on ? and then have a toggle for each of them to turn on/off the AI ?
Yes, in step 1 all of them are activated with AI ON.
The step to turn off, means disable all units (range is no longer in use). 
The AI will be handled by the IADS script activated in step 3  (So to disable, I guess first disable the IADS script from step 3, then deactivate the units)


> Step 2 you write 3 sams but there only 2 templates?
Yes, only 2 templates, but it should select randomly 3 groups based on those 2 templates, and placed at 3 locations from the list below (spawn locations).. So a total of 3 of the SPAWN locations are in in use, and it will be either a SA-6 or SA-11 there (based on the random selection)

**Comment by 132nd-Entropy** [comment]
first iteration done, WIP still

**Comment by 132ndNeck** [comment]
I have tested this a couple of times now. It seems that the workaround is not working, as it removed the Skynet functionality. The SAMs are on early and do not wait until the aircraft are within distance.

I made a version where I only rely on Skynet (but then looses the dynamic spawning on the different locations. This can be enabled by moving a veihcle on Banak airbase (picture below). This activates the units and enables the skynet script. When I do it this way the SAMs do not turn on before the aircraft are in range.
![bilde](https://github.com/user-attachments/assets/141f1e2e-499e-4f81-9c69-8cc6d842c13b)
